loader/loader.c

```diff
@@ -3516,6 +3736,23 @@ VkResult loader_icd_scan(const struct loader_instance *inst, struct loader_icd_t
     // a failure occurs before allocating the manifest filename_list.
     memset(&manifest_files, 0, sizeof(struct loader_data_files));
 
+    // Set up the ICD Trampoline list so elements can be written into it.
+    res = loader_scanned_icd_init(inst, icd_tramp_list);
+    if (res == VK_ERROR_OUT_OF_HOST_MEMORY) {
+        return res;
+    }
```

Comment:
Wasn't this originally done somewhere that was executed once per executable?  IDK, that's the murkiest area of the loader to me, TBH.

Reply:
I was thinking `loader_preload_icds`, so if I'm wrong just ignore me.

Reply:
loader_scanned_icd_init is just clearing & initializing  the array to store the ICD structures in it.
Preload icd's is very stupid in that all it does is create a separate ICD list so that the dynamic libraries get loaded once and the OS will not 'unload' in between calls to the loader.